Require two channel updates

andrei-21 opened the pull request:
In order to generate a snapshot with information about a channel, it needs two channel updates (one for each end). The change is for Regtest because usually there are only few channels.

TheBlueMatt commented on src/tracking.rs:
```diff
@@ -118,7 +120,14 @@ pub(crate) async fn download_gossip<L: Deref + Clone + Send + Sync + 'static>(pe
+			if chain_hash == ChainHash::REGTEST {
+				// There must be two channel updates (one for each end)
+				// to generate a snapshot with information about the channel.
+				is_caught_up_with_gossip = new_message_count < 20 && previous_announcement_count > 0 && previous_update_count > 0 && counter.channel_updates > 1;
```
We're definitely not done syncing mainnet if the total number of channel updates we've seen is only one :)

TheBlueMatt replied:
Yes, `counter.channel_updates` should be strictly increasing forever.

andrei-21 replied:
Ok, I removed the Regtest condition.

andrei-21 committed:
Require two channel updates

In order to generate a snapshot with information about a channel, it
needs two channel updates (one for each end).

TheBlueMatt reviewed (approved):
Thanks!
